AutoRest/Generators/CSharp/CSharp/ClientModelExtensions.cs

```diff
@@ -13,6 +13,22 @@ namespace Microsoft.Rest.Generator.CSharp
     public static class ClientModelExtensions
     {
         /// <summary>
+        /// Determine whether URL encoding should be skipped for this parameter
+        /// </summary>
+        /// <param name="parameter">The parameter to check</param>
+        /// <returns>true if url encoding should be skipped for the parameter, otherwise false</returns>
+        public static bool SkipUrlEncoding(this Parameter parameter)
```

Comment:
This is a good extension method and it should exist in the extensions. Please move it there and use it whenever we check for `Parameter.Extensions.ContainsKey(AzureExtensions.SkipUrlEncodingExtension)`.